pkgs/by-name/vt/vtsls/package.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, importNpmLock
+}:
+
+buildNpmPackage rec {
+  pname = "vtsls";
+  version = "0.2.3";
+
+  src = fetchFromGitHub {
+    owner = "yioneko";
+    repo = "vtsls";
+    rev = "server-v${version}";
+    hash = "sha256-rHiH42WpKR1nZjsW+Q4pit1aLbNIKxpYSy7sjPS0WGc=";
+    fetchSubmodules = true;
+  };
+
+  sourceRoot = "${src.name}/packages/server";
+
+  npmDeps = importNpmLock {
+    npmRoot = "${src}/packages/server";
+    packageLock = lib.importJSON ./package-lock.json;
+  };
```

Comment:
```suggestion
  npmDeps = importNpmLock {
    package = lib.importJSON ./package.json; # you need to vendor also this file
    packageLock = lib.importJSON ./package-lock.json;
  };
```

IFD is a nix feature allowed by default in nix but forbidden in nixpkgs. My change above eliminates it but introduces other errors that I wasn't able to quickly fix.
Use 
```
nix --option allow-import-from-derivation false build .#vtsls -L --no-eval-cache
```

from your nixpkgs directory to be sure that there are no IFDs (I'm also assuming flakes with the command above).
Anyway how did you get the `package-lock.json` file? Did you use `npm i --package-lock-only`? In that case it means that we are using different packages versions than upstream.

I saw that there is a package called `pnpm-lock-export` that should convert the original `pnpm-lock.yaml` file to a `package-lock.json` but it fails with that specific lockfile. There is also [pnpm-lock-to-npm-lock](https://github.com/jakedoublev/pnpm-lock-to-npm-lock) but I haven't tried since it's not in Nixpkgs. However I believe you should document in a comment the needed to command to generate the lockfile, or even better, write an update script that does it automatically.

Reply:
Thank you very much for the insight @aciceri! Yes I had just created a lockfile myself, since I couldn't create one otherwise (as far as I knew).

I'm seeing this quite late however, so now there seems to be an official way to handle workspaces as @lxsymington mentioned promptly mentioned - I'll be looking into that now and follow up the PR with it if things go smooth.